BootloaderCorePkg/Library/S3SaveRestoreLib/S3SaveRestore.c

```diff
@@ -139,6 +139,37 @@ TriggerPayloadSwSmi (
   IoWrite8(REG_APM_CNT, SwSmiNumber);
 }
 
+/**
+  This function clears TSEG area designated for S3
+  save/restore purpose.
+
+  @retval   EFI_OUT_OF_RESOURCES  If SmmSize is exceeding 4KiB
+  @retval   EFI_SUCCESS           Successful
+
+**/
+EFI_STATUS
+EFIAPI
+ClearS3SaveRegion (
+  VOID
+)
+{
+  LDR_SMM_INFO      LdrSmmInfo;
+  UINT8             *SmmBase;
+  UINT32            SmmSize;
+
+  PlatformUpdateHobInfo (&gSmmInformationGuid, &LdrSmmInfo);
+  SmmBase = (UINT8 *)(UINTN)LdrSmmInfo.SmmBase;
+  if (LdrSmmInfo.Flags & SMM_FLAGS_4KB_COMMUNICATION) {
+    SmmSize = SIZE_4KB;
+  } else {

```

Comment:
This logic would make the function simple:
if ((LdrSmmInfo.Flags & SMM_FLAGS_4KB_COMMUNICATION) != 0) {
  ZeroMem (SmmBase, SIZE_4KB);
}

Reply:
got it. thanks for the comments